### ISSUE TITLE
[@types/react-redux] Issues with typescript@3.5.0-rc

### ISSUE THREAD

**Issue by garycourt** [DefinitelyTyped/DefinitelyTyped#35695]
- [x] I tried using the `@types/react-redux@^7.0.9` package and had problems.
- [x] I tried using the latest RC version of tsc. https://www.npmjs.com/package/typescript
- [x] I have a question that is inappropriate for [StackOverflow](https://stackoverflow.com/).  (Please ask any appropriate questions there).
- [x] [Mention](https://github.com/blog/821-mention-somebody-they-re-notified) the authors (see `Definitions by:` in `index.d.ts`) so they can respond.
  - Authors: @tkqubo, @kenzierocks, @clayne11, @tansongyang, @nicholasboll, @mdibyo, @kallikrein, @val1984, @jrakotoharisoa, @apapirovski, @surgeboris, @soerenbf

I am getting the following error when using `@types/react-redux@^7.0.9` with `typescript@3.5.0-rc`:

```
node_modules/@types/react-redux/index.d.ts
(103,79): Namespace '"node_modules/hoist-non-react-statics/index"' has no exported member 'NonReactStatics'.
```

**Comment by garycourt** [comment]
At first glance, it appear that the issue is that `hoist-non-react-statics` includes it's own `index.d.ts` file, which is overriding `@types/hoist-non-react-statics` and does not include the `NonReactStatics` interface.

**Comment by garycourt** [comment]
I figured out this issue. The problem was that another dependency in my package was importing an older version of `hoist-non-react-statics`, which was getting placed in `node_modules/hoist-non-react-statics`, and `typescript@3.5.0-rc` was importing the `index.d.ts` file in this version, which was causing the type error.

I was able to solve the issue by running `yarn add hoist-non-react-statics@^3.3.0` to my package, which forces `node_modules/hoist-non-react-statics` to the correct version, and thus the `index.d.ts` file is not there to cause type issues.
